### PR TITLE
Update REAME example links for v1alpha1

### PR DESCRIPTION
**Pull request by alexandraj777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/452)

### CODE REVIEW

**Comment by alexandraj777** [comment]
Eeek, commit message typo 😱

File: README.md [hunk 1]
```diff
@@ -189,21 +189,21 @@ spec:
...
 ```
 
 This is example for pytorch operator
 
 ```
-kubectl create -f https://raw.githubusercontent.com/kubeflow/katib/master/examples/pytorchjob-example.yaml
+kubectl create -f https://raw.githubusercontent.com/kubeflow/katib/master/examples/v1alpha1/pytorchjob-example.yaml
 ```
 
 You can check status of StudyJob
```
Comment (andreyvelich) [inline README.md]: @alexandraj777 Thank you for your PR! 
Should we also update the link here: https://github.com/kubeflow/katib/pull/452/files#diff-04c6e90faac2675aa89e2176d2eec7d8R161
and here: https://github.com/kubeflow/katib/pull/452/files#diff-04c6e90faac2675aa89e2176d2eec7d8R362  ?

File: README.md [hunk 2]
```diff
@@ -189,21 +189,21 @@ spec:
 If you are not using GKE and you don't have StorageClass for dynamic volume provisioning at your cluster, you have to create pvc and pv
 
 ```
-kubectl create -f https://raw.githubusercontent.com/kubeflow/katib/master/examples/tfevent-volume/tfevent-pvc.yaml
+kubectl create -f https://raw.githubusercontent.com/kubeflow/katib/master/examples/v1alpha1/tfevent-volume/tfevent-pvc.yaml
```
Comment (hougangliu) [inline README.md]: Thanks!
please also update linke of line 93, too.
In fact, links in https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md need be updated, too

**Comment by alexandraj777** [comment]
@hougangliu @andreyvelich thanks for the review! 

Seems like there's a lot of places to update v1 links. Here, I've only updated the `examples/` links, but some of the ones you've pointed out are from the `pkg` folder and the `manifests` folder. When I do these in my own repos, I generally like to do PRs based on easy search-and-replaces, with roughly one PR per search and replace. So, I would normally do one PR for `examples/` links, one PR for replacing `pkg/` links, one for `manifests/` links. 

I'm not too familiar with how you like to do PRs in this codebase, but would it be ok with you to just leave this PR for the `examples/` links, and do followup PRs for the `pkg/` changes and the `manifests/` changes? Or is separate PRs too much load on reviewers? If not, I'm happy to add the changes to this PR.

**Comment by richardsliu** [comment]
@alexandraj777 If it's not too much trouble could you fix the other links in the same PR? It is generally cleaner and makes things easier to track. Thanks!

**Comment by alexandraj777** [comment]
@richardsliu that's reasonable, will do

**Comment by alexandraj777** [comment]
Ok I think I've caught all of the changes between `pkg`, `examples`, and `manifests`. The biggest change was rewriting parts of the `pkg/api/README.md` file to include links for both `v1alpha1` and `v1alpha2` links. I've attempted to manually verify that all links exist.

This should be ready to go! @hougangliu @andreyvelich anything I missed?

**Comment by hougangliu** [comment]
@alexandraj777 thanks for your PR.
/lgtm

**Comment by richardsliu** [comment]
Thanks Alexandra.
/approve

File: docs/developer-guide.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ You can build all images from source.
 
 ## Implement new suggestion algorithm
```
Comment (andreyvelich) [inline docs/developer-guide.md]: I think, we should also change path here: https://github.com/kubeflow/katib/pull/452/files#diff-ba23e0f18c79cb873ac5375367524cfaL20.

File: pkg/api/README.md [hunk 4]
```diff
@@ -4,10 +4,17 @@ Manager service is an API for core component.
 Suggestion service is an API for Suggestion services.
 EarlyStopping service is an API for EarlyStopping services.
 
-## Documentation
-Please refer to [api.md](./gen-doc/api.md).
+## v1alpha1 Documentation
```
Comment (andreyvelich) [inline pkg/api/README.md]: Probably, just name it `Documentation`? Since we have v1alpha1 and v1alpha2 links here.

**Comment by andreyvelich** [comment]
@alexandraj777 Thanks for your changes! I left 2 commentaries.

File: examples/v1alpha1/README.md [hunk 5]
```diff
@@ -369,7 +369,7 @@ batch2 accuracy=0.8
 Validation-accuracy=0.75
 ```
 The metrics collector will collect all logs of metrics.
-The manifest of metrics collector is also generated from template and defined [here](/manifests/studyjobcontroller/metricsControllerConfigMap.yaml).
+The manifest of metrics collector is also generated from template and defined [here](./manifests/v1alpha1/studyjobcontroller/metricsControllerConfigMap.yaml).
```
Comment (andreyvelich) [inline examples/v1alpha1/README.md]: As well, I found this line with wrong link: 
https://github.com/kubeflow/katib/pull/452/files#diff-be7f7f1b07a698ff4c53ad0e30462b67R202

File: docs/developer-guide.md (hunk 3 above)
Comment (alexandraj777) [reply docs/developer-guide.md]: done

**Comment by alexandraj777** [comment]
@andreyvelich I've made changes based on all of your comments!

**Comment by andreyvelich** [comment]
Thank you!
/lgtm
/hold cancel

**Pull request merged**